File: backend/smart-campus-api/src/main/java/com/smartcampus/service/CommentService.java
package com.smartcampus.service;

import com.smartcampus.dto.comment.CommentCreateRequest;
import com.smartcampus.dto.comment.CommentResponse;

import java.util.List;

public interface CommentService {
    CommentResponse addComment(String ticketId, CommentCreateRequest request, String authorId);
    List<CommentResponse> getCommentsByTicket(String ticketId);
    CommentResponse editComment(String commentId, String newContent, String requesterId);
    void deleteComment(String commentId, String requesterId, boolean isAdmin);
}

File: frontend/src/services/api/commentService.js
import api from './axios';

export const commentService = {
  // Add a comment to a ticket
  addComment: async (ticketId, content) => {
    const response = await api.post(`/tickets/${ticketId}/comments`, { content });
    return response.data;
  },

  // Get all comments for a ticket
  getCommentsByTicket: async (ticketId) => {
    const response = await api.get(`/tickets/${ticketId}/comments`);
    return response.data;
  },

  // Edit a comment (Owner only)
  editComment: async (ticketId, commentId, newContent) => {
    const response = await api.put(`/tickets/${ticketId}/comments/${commentId}`, { content: newContent });
    return response.data;
  },

  // Delete a comment (Owner or Admin)
  deleteComment: async (ticketId, commentId) => {
    const response = await api.delete(`/tickets/${ticketId}/comments/${commentId}`);
    return response.data;
  }
};
